### PR TITLE
INC-379: use Incentives API instead of Prison API to get incentive summary

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/INC-379

### CODE REVIEW

File: e2e/cypress/mockApis/incentivesApi.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    response: {
+      status: 200,
+      headers: {
+        'Content-Type': 'application/json;charset=UTF-8',
+      },
+      jsonBody,
+    },
+  });
+
+const stubIncentives = incentives =>
```
Comment (neilmendum) [inline e2e/cypress/mockApis/incentivesApi.js:18]: Created this before realising that all cypress tests have been moved to the `ignored` directory and the `run_e2e_tests` CircleCI job has been commented out. But at least we have a stub if they're ever switched back on

File: server/repositories/__tests__/offenderRepository.spec.js [hunk 2]
```diff
@@ -51,13 +51,19 @@ describe('offenderRepository', () => {
 
   describe('getIncentivesSummaryFor', () => {
     it('calls the Incentives summary endpoint for a given ID', async () => {
-      const client = {
```
Comment (neilmendum) [inline server/repositories/__tests__/offenderRepository.spec.js:54]: I left the other tests alone which have a single `client` passed in and stubbed

File: server/services/offender/index.js [hunk 3]
```diff
@@ -270,5 +270,5 @@ const createOffenderService = (
 };
 
 module.exports = {
-  createPrisonApiOffenderService: createOffenderService,
+  createPrisonAndIncentivesApiOffenderService: createOffenderService,
```
Comment (neilmendum) [inline server/services/offender/index.js:273]: Slightly clunky name so open to suggestions 😄

File: helm_deploy/prisoner-content-hub-frontend/templates/_envs.tpl [hunk 4]
```diff
@@ -21,13 +21,19 @@ env:
...
           key: hmppsAuthBaseUrl
-    
+
     - name: PRISON_API_BASE_URL
       valueFrom:
         secretKeyRef:
           name: {{ include "prisoner-content-hub-frontend.fullname" . }}
           key: prisonApiBaseUrl
 
+    - name: INCENTIVES_API_BASE_URL
```
Comment (neilmendum) [inline helm_deploy/prisoner-content-hub-frontend/templates/_envs.tpl:31]: We will need to add this secret to the 3 namespaces

File: server/clients/__tests__/incetivesApiClient.spec.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
+const nock = require('nock');
```
Comment (neilmendum) [inline server/clients/__tests__/incetivesApiClient.spec.js:1]: Duplicated from `prisonApiClient.spec.js`

File: server/services/offender/index.js (hunk 3 above)
Comment (andrewrlee) [reply server/services/offender/index.js:273]: `createOffenderService`

File: server/repositories/offender.js [hunk 6]
```diff
@@ -5,61 +5,72 @@ function validateOffenderNumberFor(offenderNo) {
...
+        `${prisonApiBaseUrl}/bookings/offenderNo/${offenderNo.toUpperCase()}`,
       );
     }
     throw new Error('Invalid offender number');
   }
 
   function getIncentivesSummaryFor(bookingId) {
-    return httpClient.get(`${baseUrl}/bookings/${bookingId}/iepSummary`);
+    return incentivesApiHttpClient.get(
+      `${incentivesApiBaseUrl}/iep/reviews/booking/${bookingId}/iepSummary`,
```
Comment (Mjwillis) [inline server/repositories/offender.js:23]: This URL api entry looks odd - iepSummary at the end?

File: e2e/cypress/mockApis/incentivesApi.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  stub(`/incentivesapi/iep/reviews/booking/.*?/iepSummary`, incentives);
```
Comment (Mjwillis) [inline e2e/cypress/mockApis/incentivesApi.js:19]: again that is not the URL

**Review by Mjwillis — CHANGES_REQUESTED** [review]
URL for API for incentives looks wrong

File: server/repositories/offender.js (hunk 6 above)
Comment (neilmendum) [reply server/repositories/offender.js:23]: Good spot! I was mixing up Prison API and Incentives API endpoints

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**